### PR TITLE
[RAPTOR-8526] Initialize the DataRobot Python client at the drum layer

### PR DESCRIPTION
**Pull request by zohar-mizrahi**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
In the context of allowing DataRobot custom models to access DataRobot public API, it is required to initialize the DataRobot python client at the drum layer. 

## Rationale
The idea is to make it easy for users to use DataRobot Python client in their custom models, but just importing datarobot and immediately use it without the need to initialize the client connection to DataRobot application.

### CODE REVIEW

**Review by yakov-g — APPROVED** [review]

**Comment by yakov-g** [comment]
I don't know why test is failing, `test_integration_in_a_single_container` , but after I merge my PR, it won't run in the container anymore

**Review by rvorobii — APPROVED** [review]

File: custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py [hunk 1]
```diff
@@ -38,6 +39,11 @@ def __init__(self):
     def mlpiper_configure(self, params):
         super(PythonPredictor, self).mlpiper_configure(params)
 
+        if to_bool(params.get("allow_dr_api_access")):
+            logger.info("Initializing DataRobot Python client.")
+            dr_api_endpoint = self._dr_api_url(params["external_webserver_url"])
```
Comment (akshoop) [inline custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py:44]: nit:
```suggestion
            dr_api_endpoint = self._dr_api_url(endpoint=params["external_webserver_url"])
```

**Review by akshoop — APPROVED** [review]
Overall LGTM from Core Modeling, CC @eric-s-s @scottp-ml for awareness such as new method verify_dr_api_access_options

**Comment by yakov-g** [comment]
jenkins run tnis

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Comment by yakov-g** [comment]
jenkins test this

**Comment by yakov-g** [comment]
jenkins test this

**Comment by yakov-g** [comment]
jenkins test this

**Comment by yakov-g** [comment]
jenkins run test_integration_java_environment test_integration_r_environment

**Pull request merged**